core/azure-core-auth/src/main/java/com/azure/core/auth/credentials/AzureTokenCredentials.java

```diff
@@ -84,7 +84,9 @@ public AzureTokenCredentials(AzureEnvironment environment, String domain) {
     public abstract Mono<String> getToken(String resource);
 
     @Override
-    public Mono<String> getTokenAsync(String resource) {
+    public Mono<String> getToken(String... scopes) {
+        String scope = scopes[0];
+        String resource = scope.substring(0, scope.lastIndexOf('.'));
         return getTokenFromUri(resource).map(token -> SCHEME + token);
```

Comment:
ScopeUtil is the utility class handling the scopes now. (https://github.com/Azure/azure-sdk-for-java/pull/3824/files#diff-e7bf4af51e3faed8f9ef8390eb2ca4e6). It's awkward for now since we are using adal4j.

Reply:
Moving `ScopeUtil` to `azure-core`.